src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+interface TodoModalProps {
+  onClose: (value: undefined) => void;
+  post: Todo;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  onClose,
+  post: { userId, id, title, completed },
+}) => {
+  const [user, setUser] = useState<User>();
+  const [loaded, setLoaded] = useState(false);
+
+  let name, email;
```

Comment:
The variables `name` and `email` should be initialized with default values to avoid potential undefined errors when rendering the component before the user data is loaded.